test/Frontend/dependencies-selective-supplementaries.swift

```diff
@@ -0,0 +1,15 @@
+// This test verifies that, in batch-mode (i.e. >1 primary input),
+// we have fixed a bug that caused every primary input to be a dependent
+// of every ModuleOutputPath, etc.
+
+// RUN: %empty-directory(%t)
+// RUN: echo 'public func a() { }' >%t/a.swift
+// RUN: echo 'public func main() {a()}' >%t/main.swift
+// RUN: %target-swift-frontend -c -enable-batch-mode -bypass-batch-mode-checks -module-name foo -primary-file %t/a.swift -primary-file %t/main.swift -emit-dependencies-path %t/a.d -emit-dependencies-path %t/main.d  -o a.o -o main.o -emit-module-path a.swiftmodule -emit-module-path main.swiftmodule
+// RUN: not %FileCheck -check-prefix=CHECK-MAIN %s <%t/a.d
+// RUN: not %FileCheck -check-prefix=CHECK-A %s <%t/main.d
```

Comment:
FileCheck actually has a nice way to do negative checks, better than using `not`: you can write `CHECK-A-NOT` instead of `CHECK-A` as the check line. The catch is that if you combine that with positive tests, it only checks that the match string does not appear *between the positive matches.*

I do think positive tests would be good for this, so I'd do something like

```
// CHECK-MAIN-DAG: main.swiftmodule:
// CHECK-MAIN-DAG: main.o:
// NEGATIVE-MAIN-NOT: a.swiftmodule
// NEGATIVE-MAIN-NOT: a.o
```

and then invoke FileCheck twice:

```
%FileCheck -check-prefix=CHECK-MAIN %s <%t/main.d
%FileCheck -check-prefix=NEGATIVE-MAIN %s <%t/main.d
```

(The "DAG" means "ignore the order of results until the next non-DAG line".)

Also, nitpick: make sure all your output files are going into %t as well. Otherwise it'll litter some random directory that lit picks as the working directory, or possibly even your actual working directory.

Reply:
Thanks for this help. I have fixed up the test. Are we good to go?